entry

```diff
@@ -117,7 +117,18 @@ helm_repo_init() {
 	fi
 
 	if [[ -n "${REPO}" ]]; then
-		${HELM} repo add ${CA_FILE_ARG} ${NAME%%/*} ${REPO}
+		if [[ -f "${AUTH_DIR}/username" ]] && [[ -f "${AUTH_DIR}/password" ]]; then
+			if [[ "${HELM}" == "helm_v3" ]]; then
+				cat ${AUTH_DIR}/password | ${HELM} repo add ${CA_FILE_ARG} --username "$(cat ${AUTH_DIR}/username)" --password-stdin ${NAME%%/*} ${REPO}
+			else
+				set +x
+				echo ${HELM} repo add ${CA_FILE_ARG} --username "$(cat ${AUTH_DIR}/username)" --password +REDACTED+ ${NAME%%/*} ${REPO}
+				${HELM} repo add ${CA_FILE_ARG} --username "$(cat ${AUTH_DIR}/username)" --password "$(cat ${AUTH_DIR}/password)" ${NAME%%/*} ${REPO}
```

Comment:
From our offline discussion, giving that Helm v2 is already EOL and doesn't support `--password-stdin`, could we not support this feature in v2? Would it be feasible? Otherwise we can/could document this specific behavior in the docs to alert users.

Reply:
With this change the password is not logged.

Reply:
Agree with your analysis 👍🏻

Reply:
I waffled a bit and decided to just not support it for v2. There is another setting that users may want that isn't supported by helm v2 so lets just not do it at all.